packages/arb-token-bridge-ui/src/util/TokenUtils.ts

```diff
@@ -365,3 +366,38 @@ export function erc20DataToErc20BridgeToken(data: Erc20Data): ERC20BridgeToken {
     listIds: new Set()
   }
 }
+
+export async function isCustomGatewayRegistered({
+  erc20ParentChainAddress,
+  parentChainProvider,
+  childChainProvider
+}: {
+  erc20ParentChainAddress: string
+  parentChainProvider: Provider
+  childChainProvider: Provider
+}): Promise<boolean> {
+  const erc20Bridger = await Erc20Bridger.fromProvider(childChainProvider)
+
+  const tokenChildChainAddressFromParentGatewayRouter =
+    await erc20Bridger.getL2ERC20Address(
+      erc20ParentChainAddress,
+      parentChainProvider
+    )
+
+  const childChainGatewayAddressFromChildChainRouter =
+    await erc20Bridger.getL2GatewayAddress(
+      erc20ParentChainAddress,
+      childChainProvider
+    )
+
+  const tokenChildChainAddressFromChildChainGateway =
+    await L2ERC20Gateway__factory.connect(
+      childChainGatewayAddressFromChildChainRouter,
+      childChainProvider
+    ).calculateL2TokenAddress(erc20ParentChainAddress)
```

Comment:
you can call `calculateL2TokenAddress` on child router in 1 call instead of 2

Reply:
as discussed, the sdk doesn't expose the gateway router contract for calling

Reply:
fair, let's consider moving this to sdk later